website/content/labs/lab6.md

```diff
@@ -0,0 +1,176 @@
+
+There are actually many other options you can set in `.gitlab-ci.yml`. We will only cover a few others in this lab, but you can see the complete reference here: https://docs.gitlab.com/ee/ci/yaml/
+
+Now, do the following:
+
+* [0 points] Commit the `.gitlab-ci.yml` to your repository and push it.
+
+* [5 points] Go to your repository on the GitLab server (i.e., https://mit.cs.uchicago.edu/cmsc22000-spr-18/CNETID, where CNETID should be replaced by your CNetID). On the left sidebar, click on “Pipelines”. This will show you the pipelines that have been run in your repository. You should see a “passed” pipeline (if you do not, please ask for help). Click on either “passed” or the pipeline number. This will take you to a page with more details about the pipeline. Copy the URL of that page and paste it in the `tasks.txt` file in `lab6/`. This page shows more details about the jobs that ran in that pipeline. In this case, we have only one job: `hello:build`. If you click on it, you’ll be able to see the terminal output from the job (which was run on a special server known as the build server)
+
+* [5 points] Make a change to the `hello.c` file that will make it not build. Commit and push this change to the repository. This will result in a failed pipeline in the “Pipelines” page; copy-paste the URL of the failed pipeline into the `tasks.txt` file. Before moving on to the next task, undo the change and verify that pushing the code results in a “passed” pipeline.
+
+* [15 points] Add a job called `libgeometry:build` to build the libgeometry library contained in `lab6/libgeometry`. Push your updated `.gitlab-ci.yml` file, and copy-paste the URL of the resulting “passed” pipeline into the `tasks.txt` file. Notice how this pipeline now has two jobs in it.
+
+Note: Before committing your `.gitlab-ci.yml` file, you can check that the format of the file is correct by going to https://mit.cs.uchicago.edu/ci/lint. This will allow you to copy-paste your `.gitlab-ci.yml` file and check it for syntax errors.
+
+
+
+# Task 2: Adding tests
+
+As we mentioned previously, CI typically involves running tests every time we push changes to the repository. This means that, besides building our code, we also want tests to run automatically. For the simple hello program, we’ve included a very simple test: we simply run the program (using the `tests` target in the Makefile). While this doesn’t guarantee that the program prints the correct output, it will at least verify that the program runs without errors.
+
+So, let’s add a job to the `.gitlab-ci.yml` file to run the tests:
+
+    hello:test:
+        stage: test
+        script:
+          - make -C lab6/hello clean tests
+        only: 
+          - master
+
+Notice how the stage is `test`. This means that GitLab will not run this job unless all the jobs in the `build` stage succeed.
```

Comment:
This is probably worth explaining in more depth. We should add a sentence on line 60 (where stages are introduced) explaining that stages depend on previous stages building successfully.